### PR TITLE
Replace use of JsonObject with JsonNode for CruiseControlApi

### PR DESCRIPTION
**Pull request by tinaselenge**
### Type of change

_Select the type of your PR_

- Refactoring

### Description

This PR replaces Vertx's JsonObject used for parsing CC API responses with Jackson's JsonNode. So that Vertx is completely removed from CC API. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
Thanks. LGTM.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java [hunk 1]
```diff
@@ -67,6 +71,7 @@ public CruiseControlApiImpl(int idleTimeout, Secret ccSecret, Secret ccApiSecret
         this.authHttpHeader = getAuthHttpHeader(apiAuthEnabled, ccApiSecret);
         this.pemTrustSet = new PemTrustSet(ccSecret);
         this.httpClient = buildHttpClient();
+        this.objectMapper = new ObjectMapper();
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:74]: Wouldn't a static `ObjectMapper` instance as you did for the `KafkaRebalanceAssemblyOperator` be better?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java [hunk 2]
```diff
@@ -179,23 +192,36 @@ private CompletableFuture<CruiseControlRebalanceResponse> internalRebalance(Reco
...
+                            json = objectMapper.readTree(response.body());
+                        } catch (JsonProcessingException e) {
+                            return CompletableFuture.failedFuture(new CruiseControlRestException(
+                                    "Error for request: " + host + ":" + port + path + ". Failed to deserialize the response: " + e));
+                        }
+
+                        userTaskID = response.headers().firstValue(USER_TASK_ID_HEADER).orElse("");
+
+                        if (json.has(CC_REST_API_ERROR_KEY)) errorMessage = json.get(CC_REST_API_ERROR_KEY).asText();
+                    }
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:211]: I understood what you are trying to achieve with the above code. You are "centralising" where the JSON is extracted from the response for the several status codes. The drawbacks I see is that if we need to handle an additional `statusCode`, we always need to add the new code here as well (and not just later on where we have the `if` branch for handling its logic).
I was wondering if it's better factoring out a private method to contain the code which extract the JSON from the response or returning the `CruiseControlRestException` exception if any error.
Using this method in all the places where we actually have `JsonObject json = new JsonObject(response.body());` and leaving the rest of the code the same as it's currently. Would it work in your opinion?

Comment (tinaselenge) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java]: That is a good point. I took your suggestion made an update. Thank you Paolo.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java [hunk 3]
```diff
@@ -179,23 +195,22 @@ private CompletableFuture<CruiseControlRebalanceResponse> internalRebalance(Reco
                     // send request and handle response
                     LOGGER.traceCr(reconciliation, "Response: {}, body: {}", response, response.body());
                     int statusCode = response.statusCode();
+                    String userTaskID = response.headers().firstValue(USER_TASK_ID_HEADER).orElse("");
+
                     if (statusCode == 200 || statusCode == 201) {
-                        String userTaskID = response.headers().firstValue(CruiseControlHeaders.USER_TASK_ID_HEADER).orElse("");
-                        JsonObject json = new JsonObject(response.body());
-                        LOGGER.debugCr(reconciliation, "Got {} response to POST request to {} : userTaskID = {}", response.statusCode(), path, userTaskID);
```
[the remark this answers is not in the document]
Comment (tinaselenge) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:185]: oops, that's a mistake. Thanks for spotting it. I will add it back and make sure I didn't remove any of the other logs.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!

**Comment by scholzj** [comment]
Thanks for the PR @tinaselenge

**Pull request merged**